Directory.Packages.props

```diff
@@ -52,23 +53,23 @@
         3) The .NET runtime team provides extra backward compatibility guarantee to System.Diagnostics.DiagnosticSource
           even during major version bumps, so compatibility is not a concern here.
     -->
-    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-rc.1.24431.7" />
+    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="$(LatestRuntimeOutOfBandVer)" />
+  </ItemGroup>
 
-    <!-- A conservative version of System.Text.Encodings.Web must be used here since there is no backward compatibility guarantee during major version bumps. -->
+  <ItemGroup>
+    <!-- We use conservative versions of these packages for older runtimes where an upgrade might introduce breaking changes -->
```

Comment:
@CodeBlanch, if you are keeping it legacy only for AutoInstrumentation purposes - it is not the case. The bigger problem we have with upgrading packages for the .NET6+.

From this perspective I would keep it consistent and upgrade to 8.0.0/8.0.4 respectively both for .NET Framework and .NET. Then avoid any other upgrades until we really need this. So still be convervative.

If possible, it will be great to release together with other dependency-breaking changes like System.Diagnostics.DiagnosticSource 9 and avoiding patch release with only this bump.

Reply:
> If possible, it will be great to release together with other dependency-breaking changes like System.Diagnostics.DiagnosticSource 9 and avoiding patch release with only this bump.

I'm a bit confused here, I think we've been clear about `System.Diagnostics.DiagnosticSource` - there is *no* breaking change during major version bumps.

Reply:
It is from the auto instrumentation users. Any common package changes can be problematic.

Reply:
I see, that is a technical issue in the current auto-instrumentation design, we shouldn't mix it up and get ourselves confused by saying "dependency-breaking changes like System.Diagnostics.DiagnosticSource 9".

Reply:
I tweaked it so our minimum version for System.Text.Json is v8.0.4. I decided to go that direction because there were no objections from the auto-instrumentation folks and doing so allowed us to also drop explicit references to System.Text.Encodings.Web which was a nice/welcome side-effect.

Reply:
> I tweaked it so our minimum version for System.Text.Json is v8.0.4. I decided to go that direction because there were no objections from the auto-instrumentation folks and doing so allowed us to also drop explicit references to System.Text.Encodings.Web which was a nice/welcome side-effect.

This is a red flag to me - the fact that auto-instrumentation folks are fine and it will be easier for OTel .NET maintainers cannot justify such change. @CodeBlanch could you evaluate the impact and share the results here?

Reply:
I've started a discussion with the STJ team to understand the impact of upgrading vs the risk of staying on v4.7.2 (which is no longer being maintained). Waiting to hear back.

In the meantime I've reverted the version bump where it isn't strictly necessary. Also updated the description with more details.